client/my-sites/email/email-providers-comparison/in-depth/select-button.tsx

```diff
@@ -0,0 +1,62 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { hasGSuiteSupportedDomain } from 'calypso/lib/gsuite';
+import { GOOGLE_WORKSPACE_PRODUCT_TYPE } from 'calypso/lib/gsuite/constants';
+import { IntervalLength } from 'calypso/my-sites/email/email-providers-comparison/interval-length';
+import canUserPurchaseGSuite from 'calypso/state/selectors/can-user-purchase-gsuite';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { SelectButtonProps } from 'calypso/my-sites/email/email-providers-comparison/in-depth/types';
+import type { ReactElement } from 'react';
+
+const usePlanAvailable = (
+	emailProviderSlug: string,
+	intervalLength: IntervalLength,
+	selectedDomainName: string
+) => {
+	const selectedSite = useSelector( getSelectedSite );
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSite?.ID ) );
+	const domain = getSelectedDomain( {
+		domains,
+		selectedDomainName: selectedDomainName,
+	} );
+
+	const canPurchaseGSuite = useSelector( canUserPurchaseGSuite );
+
+	if ( emailProviderSlug !== GOOGLE_WORKSPACE_PRODUCT_TYPE ) {
+		return true;
+	}
+
+	if ( ! canPurchaseGSuite || ! hasGSuiteSupportedDomain( [ domain ] ) ) {
+		return false;
+	}
+
+	return intervalLength === IntervalLength.ANNUALLY;
+};
```

Comment:
Custom hook 🤙!

Reply:
That's awesome! But I'm leaning to think that maybe this function should be placed somewhere else more accesible from other code paths?

Something like: `my-sites/email/hooks/use-plan-available.tsx` as it is done for example in:
`/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/index.tsx`

Reply:
That's something I can look into in another iteration 👍.